homeassistant/components/aurora_abb_powerone/config_flow.py

```diff
@@ -0,0 +1,148 @@
+        ver = client.version()
+        partnum = client.pn()
+        ret[ATTR_MODEL] = f"{ver} ({partnum})"
+        ret[ATTR_FIRMWARE] = client.firmware(1)
+        _LOGGER.info("Returning device info=%s", ret)
+    except AuroraError as err:
+        _LOGGER.warning("Could not connect to device=%s", comport)
+        if populate_on_fail:
+            ret = {
+                "title": DEFAULT_INTEGRATION_TITLE,
+                "serial_number": "735492",
+                "pn": "-3G97-",
+                "firmware": "C.0.3.5",
+            }
+            _LOGGER.warning("Using dummy device info=%s", ret)
+        else:
+            raise err
+    finally:
+        if client.serline.isOpen():
+            client.close()
+
+    # Return info we want to store in the config entry.
+    return ret
+
+
+class AuroraABBConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Aurora ABB PowerOne."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+    _comportslist = None
+    _defaultcomport = None
+
+    def __init__(self):
+        """Initialise the config flow."""
+        self.config = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialised by the user."""
+
+        errors = {}
+        if self._comportslist is None:
+            comports = serial.tools.list_ports.comports(include_links=True)
+            comportslist = []
+            for port in comports:
+                comportslist.append(port.device)
+                _LOGGER.debug("COM port option: %s", port.device)
+            if len(comportslist) > 0:
+                defaultcomport = comportslist[0]
+            else:
+                _LOGGER.warning(
+                    "No com ports found.  Need a valid RS485 device to communicate."
+                )
+                defaultcomport = "No COM ports found"
+                errors["base"] = "no_serial_ports"
+            self._comportslist = comportslist
+            self._defaultcomport = defaultcomport
+
+        # Handle the initial step.
+        if user_input is not None:
+            try:
+                info = validate_and_connect(
+                    self.hass, user_input, user_input.get(CONF_USEDUMMYONFAIL, False)
+                )
+                info.update(user_input)
+                # Bomb out early if someone has already set up this device.
+                device_unique_id = info["serial_number"]
+                await self.async_set_unique_id(device_unique_id)
+                self._abort_if_unique_id_configured()
+
+                return self.async_create_entry(title=info["title"], data=info)
+
+            except OSError as error:
+                if "no such device" in str(error):
+                    errors["base"] = "invalid_serial_port"
+            except AuroraError as error:
+                if "could not open port" in str(error):
+                    _LOGGER.error("Unable to open serial port")
```

Comment:
Why are you logging these? You're already showing it to the user.

Reply:
Fixed.